dbms/src/Storages/Transaction/TiDB.cpp

```diff
@@ -777,14 +782,23 @@ catch (const Poco::Exception & e)
 ///////////////////////
 ////// TableInfo //////
 ///////////////////////
-TableInfo::TableInfo(Poco::JSON::Object::Ptr json)
+TableInfo::TableInfo(Poco::JSON::Object::Ptr json, KeyspaceID keyspace_id_)
 {
     deserialize(json);
+    if (keyspace_id == NullspaceID)
+    {
+        keyspace_id = keyspace_id_;
+    }
 }
 
-TableInfo::TableInfo(const String & table_info_json)
+TableInfo::TableInfo(const String & table_info_json, KeyspaceID keyspace_id_)
 {
     deserialize(table_info_json);
+    // If the table_info_json has no keyspace id, we use the keyspace_id_ as the default value.
+    if (keyspace_id == NullspaceID)
+    {
+        keyspace_id = keyspace_id_;
+    }
```

Comment:
Why not always set `keyspace_id = keyspace_Id_` ? Is there any case that you will create a TableInfo without passing a valid `keyspace_id_` when using api_v2?

Reply:
Make sense, this logic intends to use the original keyspace id if the JSON object parameter has a keyspace id field. But this might useless since there is no keyspace id in the upstream table schema.